src/main/java/org/jacksonlaboratory/service/TermService.java

```diff
@@ -48,7 +52,28 @@ public List<OntologyTerm> getAllOntologyTerms(){
 	}
 
 	public List<OntologyTerm> searchOntologyTerm(String q){
-		return this.termRepository.search(q).stream().sorted(Comparator.comparing((OntologyTerm term) -> !term.getName().startsWith(q))
-						.thenComparing(OntologyTerm::getName)).collect(Collectors.toList());
+		if (q.toUpperCase().startsWith(String.format("%s:", ontologyName.toUpperCase()))){
+			return this.termRepository.search(q, true).stream()
+					.sorted((a, b) -> {
+						int c = Integer.parseInt(a.getTermId().getId());
```

Comment:
This will fail for some ontology classes where `id` is not an integer, e.g. `NCIT:C2852` for [Adenocarcinoma](http://purl.obolibrary.org/obo/NCIT_C2852) (if you think of this serving such an ontology).

To best of my knowledge, both CURIE prefix and id are strings, but some `id`s are strings of int values.

Reply:
What's your reccomendation here string compare?

Reply:
String compare is probably the simplest solution. Any other sorting strategy should be motivated by a specific requirements. Therefore, if you are aware of no such requirements (I am not), doing string comparison sounds OK to me.